### ISSUE TITLE
chan direction being switched

### ISSUE THREAD

**Issue by lkysow** [petergtz/pegomock#79]
If I have an interface:
```go
type MyInt interface{
     MyFunc() (<-chan string, <-chan error)
}
```

My mock ends up with:
```go
func (mock *MockMyInt) MyFunc(l) (chan<- string, chan<- error) {
	if mock == nil {
		panic("mock must not be nil. Use myMock := NewMockClient().")
	}
	params := []pegomock.Param{log, path, args, v, workspace}
	result := pegomock.GetGenericMockFrom(mock).Invoke("RunCommandAsync", params, []reflect.Type{reflect.TypeOf((*chan<- string)(nil)).Elem(), reflect.TypeOf((*chan<- error)(nil)).Elem()})
	var ret0 chan<- string
	var ret1 chan<- error
	if len(result) != 0 {
		if result[0] != nil {
			ret0 = result[0].(chan<- string)
		}
		if result[1] != nil {
			ret1 = result[1].(chan<- error)
		}
	}
	return ret0, ret1
}
```

Which causes errors:
```
		have MyFunc() (chan<- string, chan<- error)
		want MyFunc() (<-chan string, <-chan error)
```

**Comment by petergtz** [comment]
@lkysow Looks like an ugly bug, yes. And you can tell that I'm not using channels very often ;-). Will check when I can work on this. Presumably sometime in the next 2 weeks. Meanwhile, feel free to provide a PR if you need it earlier. I'm happy to merge this.

Thanks for reporting this!

**Comment by petergtz** [comment]
@lkysow Okay, so you actually made me curious why this is not working :-D. And so I've fixed it right away. Can you please try the `develop` branch to see if it fixes your issues. If so, I'll make a new release on master with the change. Thanks.

**Comment by lkysow** [comment]
👍 haha thanks! I'll check it out and get back to you.

**Comment by lkysow** [comment]
I've confirmed that develop has the bug fixed.

**Comment by petergtz** [comment]
Cool. Thanks for confirming. Will release it on master over the next few days (no time right now :-( ).

**Comment by lkysow** [comment]
No rush, I have a workaround in place.

**Comment by petergtz** [comment]
Released in v2.1.0. Closing.
